datajunction-server/datajunction_server/api/helpers.py

```diff
@@ -687,6 +686,7 @@ async def build_sql_for_multiple_metrics(  # pylint: disable=too-many-arguments,
             )
             for col in dimension_columns
         ]
+        engine = materialized_cube_catalog.engines[0]
```

Comment:
In the case where the query is being built using the materialized table, we should pick one of the engines associated with the materialized table's catalog. For example, if the materialized table is in the Druid catalog, we should be picking Druid as the execution engine as well. This was the missing piece to actually executing some of our queries quickly.

Reply:
I think we used to have a cost associated with that, such that Druid would be picked first if available.